### PR TITLE
Implement generator improvements in #295

### PR DESCRIPTION
**Pull request by Xyene**

### CODE REVIEW

File: dmoj/executors/mixins.py [hunk 1]
```diff
@@ -111,9 +111,11 @@ def get_address_grace(self):
             def get_env(self):
                 env = {'LANG': 'C'}
                 if self.unbuffered:
-                    unbuf = self._file('unbuf.so')
-                    shutil.copyfile(nobuf_path, unbuf)
+                    unbuf = self._file('setbufsize.so')
+                    shutil.copyfile(setbufsize_path, unbuf)
                     env['LD_PRELOAD'] = unbuf
+                    env['CPTBOX_STDOUT_BUFFER_SIZE'] = 0
+                    print(env)
```
Comment (quantum5) [inline dmoj/executors/mixins.py]: Remove SVP.

File: dmoj/utils/setbufsize.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+void _DMOJ_setbuffer(FILE *handle, char *env_str) {
+    char *buf_env_str = getenv(env_str);
+    if (buf_env_str != NULL) {
+        char *end;
+        unsigned long buf_size = strtoul(buf_env_str, &end, 10);
+        if (*end == '\0') {
+            // Need this branch since IOFBF when buf_size=0 actually
+            // sets the buffer size to an OS-preferred, nonzero buffer
+            if (buf_size) {
```
Comment (quantum5) [inline dmoj/utils/setbufsize.c]: I like how you compared with NULL above but not zero here. Be consistent!

File: dmoj/executors/mixins.py [hunk 3]
```diff
@@ -111,9 +111,11 @@ def get_address_grace(self):
             def get_env(self):
                 env = {'LANG': 'C'}
                 if self.unbuffered:
-                    unbuf = self._file('unbuf.so')
-                    shutil.copyfile(nobuf_path, unbuf)
+                    unbuf = self._file('setbufsize.so')
+                    shutil.copyfile(setbufsize_path, unbuf)
                     env['LD_PRELOAD'] = unbuf
+                    env['CPTBOX_STDOUT_BUFFER_SIZE'] = 0
```
Comment (quantum5) [inline dmoj/executors/mixins.py]: Might want to provide an interface to set this.

Comment (Xyene) [reply dmoj/executors/mixins.py]: Goal was to get nobuffer functionality working first

**Pull request merged**